### DISCUSSION TITLE
Checking Concept Drift in SNARIMAX model

### DISCUSSION THREAD

**Discussion by XaviMartiCamarasa** [online-ml/river#1097] [Q&A] [answered]
Hi, I'm trying/playing/struggling in Time Series Forecasting, trying SNARIMAX model and thinking how to include Concept Drift checking.
```py
import calendar
 import math
 import datetime as dt
 from river import compose
 from river import linear_model
 from river import optim
 from river import preprocessing
 horizon = 6
 def get_month_distances(x):
     return {
         calendar.month_name[month]: math.exp(-(x['month'].month - month) ** 2)
         for month in range(1, 13)
     }

 def get_ordinal_date(x):
     return {'ordinal_date': x['month'].toordinal()}

 extract_features = compose.TransformerUnion(
     get_ordinal_date,
     get_month_distances
 )
 future = [
    {'month': dt.date(year=1961, month=m, day=1)}
        for m in range(1, horizon + 1)
 ]
 model = (
     extract_features |
     time_series.SNARIMAX(
         p=1,
         d=0,
         q=0,
         m=12,
         sp=3,
         sq=6,
         regressor=(
             preprocessing.StandardScaler() |
             linear_model.LinearRegression(
                 intercept_init=110,
                 optimizer=optim.SGD(0.01),
                 intercept_lr=0.3
             )
         )
     )
 )
 #metrica=metrics.MAE()
 for x, y in datasets.AirlinePassengers():
     print("y",y)
     print ("X",x)
     model = model.learn_one(x, y) #<<<<<How could I check here if a concept drift occurs and restart de model
     a=model.forecast(6)
     
     print ("a",a)

```
Many thanks!

**Comment by MaxHalford** [comment, marked as answer]
That's a $100k question! We have a `drift` module to check for concept drift on univariate data. But we don't good examples of using concept drift + model restarting in conjunction. We have all the pieces of the puzzle, but the puzzle hasn't been assembled yet. For instance, there is a `model.clone()` method which is handy for this.

*Small tip: you can highlight your Python code properly if you use ```py instead of just ```*

**Reply by XaviMartiCamarasa** [reply]
I was playing with clone() but I got errors like the below 

![image](https://user-images.githubusercontent.com/117338671/203755653-18bde06e-54c4-40d0-97f0-b01266b93f57.png)

Error (because of Standard Scaler used in the Linear Regressor by default):
![image](https://user-images.githubusercontent.com/117338671/203756379-e6909f53-176d-4a74-b53a-f5d65df0e37b.png)

But if I'm not wrong I could get rid off that error just modifying the model and using a pipeline like below
![image](https://user-images.githubusercontent.com/117338671/203758940-daa1ab27-2c59-4c9b-920f-06c81dc1abec.png)

Is that true? 
Many thanks!

**Reply by XaviMartiCamarasa** [reply]
If I remove the StadardScaler, the model does not forecast. The predicctions are nan!
![image](https://user-images.githubusercontent.com/117338671/203773899-b381d9cc-9895-47a7-bd62-ddb2aae5c337.png)

**Reply by XaviMartiCamarasa** [reply]
Any help with this?

**Reply by MaxHalford** [reply]
Checking now. Sorry but I also have a paying job to do :)

**Reply by MaxHalford** [reply]
Ok I found the issue. You in fact encountered a bug. I've fixed it in #1099. You'll have to install the development version if you want to benefit from it.

**Reply by XaviMartiCamarasa** [reply]
@MaxHalford , how can I do that? I'm a beginner on this!
perhaps with pip install git+https://github.com/online-ml/river --upgrade?

**Reply by MaxHalford** [reply]
>  perhaps with `pip install git+https://github.com/online-ml/river --upgrade`?

Yes exactly that :). Beware that you will need to have Rust installed.

**Reply by XaviMartiCamarasa** [reply]
Morning Max.
What is.. exactly Rust?

**Reply by XaviMartiCamarasa** [reply]
How could I get the new version of River with the fix?

**Reply by MaxHalford** [reply]
It's a compiled programming language. Installation instructions are [here](https://www.rust-lang.org/tools/install).

**Reply by XaviMartiCamarasa** [reply]
But...what is the relation between RUST and upgrading river from git?

**Reply by MaxHalford** [reply]
River has some Rust code in it. So when you install it from git (via pip), you need Rust to compiles these pieces of code. If you try without Rust, it will ask you to install Rust.

**Comment by XaviMartiCamarasa** [comment]
Your job is more important Max.
Many thanks!

On Thu, Nov 24, 2022 at 8:15 PM Max Halford ***@***.***>
wrote:

> Checking now. Sorry but I also have a paying job to do :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/online-ml/river/discussions/1097#discussioncomment-4230361>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A37HEL7EUPVQLZIU574RLFTWJ65EJANCNFSM6AAAAAASKADCRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 



INFORMACIÓ SOBRE PROTECCIÓ DE DADES DE LA UNIVERSITAT OBERTA DE 
CATALUNYA (UOC)

Us informem que les vostres dades identificatives i les 
contingudes en els missatges electrònics i fitxers adjunts es poden 
incorporar a les nostres bases de dades amb la finalitat de gestionar les 
relacions i comunicacions vinculades a la UOC, i que es poden conservar 
mentre es mantingui la relació. Si ho voleu, podeu exercir el dret a 
accedir a les vostres dades, rectificar-les i suprimir-les i altres drets 
reconeguts normativament adreçant-vos a l'adreça de correu emissora o a 
***@***.*** ***@***.***>.

Aquest missatge i qualsevol 
fitxer que porti adjunt, si escau, tenen el caràcter de confidencials i 
s'adrecen únicament a la persona o entitat a qui s'han enviat.

Així 
mateix, posem a la vostra disposició un delegat de protecció de dades que 
no només s'encarregarà de supervisar tots els tractaments de dades de la 
nostra entitat, sinó que us podrà atendre per a qualsevol qüestió 
relacionada amb el tractament de dades. La seva adreça de contacte és 
***@***.*** ***@***.***>.
INFORMACIÓN SOBRE PROTECCIÓN DE DATOS DE 
LA UNIVERSITAT OBERTA DE CATALUNYA (UOC)
Os informamos de que vuestros 
datos identificativos y los contenidos en los mensajes electrónicos y 
ficheros adjuntos pueden incorporarse a nuestras bases de datos con el fin 
de gestionar las relaciones y comunicaciones vinculadas a la UOC, y de que 
pueden conservarse mientras se mantenga la relación. Si lo deseáis, podéis 
ejercer el derecho a acceder a vuestros datos, rectificarlos y suprimirlos 
y otros derechos reconocidos normativamente dirigiéndoos a la dirección de 
correo emisora o a ***@***.*** ***@***.***>.
Este mensaje y 
cualquier fichero que lleve adjunto, si procede, tienen el carácter de 
confidenciales y se dirigen únicamente a la persona o entidad a quien se 
han enviado.
Así mismo, ponemos a vuestra disposición a un delegado de 
protección de datos que no solo se encargará de supervisar todos los 
tratamientos de datos de nuestra entidad, sino que podrá atenderos para 
cualquier cuestión relacionada con el tratamiento de datos. Su dirección de 
contacto es ***@***.*** ***@***.***>.


UNIVERSITAT OBERTA DE 
CATALUNYA (UOC) DATA PROTECTION INFORMATION
Your personal data and the data 
contained in your email messages and attached files may be stored in our 
databases for the purpose of maintaining relations and communications 
linked to the UOC, and the data may be stored for as long as these 
relations and communications are maintained. If you so wish, you can 
exercise your rights to access, rectification and erasure of your data, and 
any other legally held rights, by writing to the sender’s email address or 
***@***.*** ***@***.***>.
This message and, where 
applicable, any attachments are confidential and addressed solely to the 
individual or organization they were sent to.
The UOC has a data protection 
officer who not only supervises the data processing carried out at the 
University, but who will also respond to any questions you may have about 
this data processing. You can contact our data protection officer by 
writing to ***@***.*** ***@***.***>.

**Comment by XaviMartiCamarasa** [comment]
Let me explain what I'm trying to do. (perhaps it is better)
My idea is testing two SNARIMAX Models (with two different horizons) agains a baseline forecaster (Holtwinters as you told me).
At same time, I'd like to check if Concept Drift appears, then reset the model and keep "training" with the new fresh data.

For testing purposes, I think`iter_valuate` method from `time_series `is enough. But how introduce there the idea of `"checking concept drift"`
If I'm not mistaken, this code checks the performance of the model and I could plot the measurements:
```py
def get_month_distances(x):
    return {
        calendar.month_name[month]: math.exp(-(x['month'].month - month) ** 2)
        for month in range(1, 13)
    }

def get_ordinal_date(x):
    return {'ordinal_date': x['month'].toordinal()}

extract_features = compose.TransformerUnion(
    get_ordinal_date,
    get_month_distances
)
model_s = (
     extract_features |
     time_series.SNARIMAX(
         p=1,
         d=0,
         q=0,
         m=12,
         sp=3,
         sq=6,
         regressor=(
             preprocessing.StandardScaler() | #<<<<  Quitando el Scaler funcionar
             linear_model.LinearRegression(
                 intercept_init=110,
                 optimizer=optim.SGD(0.01),
                 intercept_lr=0.3
             )
         )
     )
 )
dataset = datasets.AirlinePassengers()
metricas = time_series.iter_evaluate(
     dataset=datasets.AirlinePassengers(),
     model=model_s,
     metric=metrics.MAE(),
     horizon=3)

for m in metricas:
       print(m)
```
Thanks in advance

**Reply by MaxHalford** [reply]
I'm not sure exactly what your question is, because you mention a few things. As I told you, there is currently no canonical way to combine a drift detection module with a predictive model. We simply haven't implemented that yet.

One thing to be aware of though, is that online machine learning models naturally forget the past. This is down to the way they learn. It's mechanical. Therefore, you usually don't have to worry too much about drift. Drift is usually more of an issue with batch models. But with online models, the more you learn, the more you forget the past. Does that make sense?

**Reply by XaviMartiCamarasa** [reply]
For Concept Drift the fastest you adapt to de Drift, the better the model will be,won't it?
Of course in the long term the model will adapt to the data, but if we do not handle the drift, the model will adapt later to the new data.
If I find a drift, I will clone the model being retrained from the "drift time"

that's my opinion

**Reply by MaxHalford** [reply]
Right, but machine learning is not about opinions. I suggest you measure the performance of both approaches to be sure.

My "opinion" is that recreating a model from scratch is detrimental in that you lose all the information you had accumulated.

I think you should distinguish slow drift from brutal drift. If the drift is brutal, then yes you might want to start from scratch. But if the drift is gradual, then you might be ok just having your model learn.

**Reply by XaviMartiCamarasa** [reply]
Any idea of how to measure the performance of both approaches?

**Comment by XaviMartiCamarasa** [comment]
Thanks for the advice.

On Thu, Nov 24, 2022 at 10:44 PM Max Halford ***@***.***>
wrote:

> Right, but machine learning is not about opinions. I suggest you measure
> the performance of both approaches to be sure.
>
> My "opinion" is that recreating a model from scratch is detrimental in
> that you lose all the information you had accumulated.
>
> I think you should distinguish slow drift from brutal drift. If the drift
> is brutal, then yes you might want to start from scratch. But if the drift
> is gradual, then you might be ok just having your model learn.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/online-ml/river/discussions/1097#discussioncomment-4232147>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A37HEL6FWXRERD55KLVYXZ3WJ7OUZANCNFSM6AAAAAASKADCRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 



INFORMACIÓ SOBRE PROTECCIÓ DE DADES DE LA UNIVERSITAT OBERTA DE 
CATALUNYA (UOC)

Us informem que les vostres dades identificatives i les 
contingudes en els missatges electrònics i fitxers adjunts es poden 
incorporar a les nostres bases de dades amb la finalitat de gestionar les 
relacions i comunicacions vinculades a la UOC, i que es poden conservar 
mentre es mantingui la relació. Si ho voleu, podeu exercir el dret a 
accedir a les vostres dades, rectificar-les i suprimir-les i altres drets 
reconeguts normativament adreçant-vos a l'adreça de correu emissora o a 
***@***.*** ***@***.***>.

Aquest missatge i qualsevol 
fitxer que porti adjunt, si escau, tenen el caràcter de confidencials i 
s'adrecen únicament a la persona o entitat a qui s'han enviat.

Així 
mateix, posem a la vostra disposició un delegat de protecció de dades que 
no només s'encarregarà de supervisar tots els tractaments de dades de la 
nostra entitat, sinó que us podrà atendre per a qualsevol qüestió 
relacionada amb el tractament de dades. La seva adreça de contacte és 
***@***.*** ***@***.***>.
INFORMACIÓN SOBRE PROTECCIÓN DE DATOS DE 
LA UNIVERSITAT OBERTA DE CATALUNYA (UOC)
Os informamos de que vuestros 
datos identificativos y los contenidos en los mensajes electrónicos y 
ficheros adjuntos pueden incorporarse a nuestras bases de datos con el fin 
de gestionar las relaciones y comunicaciones vinculadas a la UOC, y de que 
pueden conservarse mientras se mantenga la relación. Si lo deseáis, podéis 
ejercer el derecho a acceder a vuestros datos, rectificarlos y suprimirlos 
y otros derechos reconocidos normativamente dirigiéndoos a la dirección de 
correo emisora o a ***@***.*** ***@***.***>.
Este mensaje y 
cualquier fichero que lleve adjunto, si procede, tienen el carácter de 
confidenciales y se dirigen únicamente a la persona o entidad a quien se 
han enviado.
Así mismo, ponemos a vuestra disposición a un delegado de 
protección de datos que no solo se encargará de supervisar todos los 
tratamientos de datos de nuestra entidad, sino que podrá atenderos para 
cualquier cuestión relacionada con el tratamiento de datos. Su dirección de 
contacto es ***@***.*** ***@***.***>.


UNIVERSITAT OBERTA DE 
CATALUNYA (UOC) DATA PROTECTION INFORMATION
Your personal data and the data 
contained in your email messages and attached files may be stored in our 
databases for the purpose of maintaining relations and communications 
linked to the UOC, and the data may be stored for as long as these 
relations and communications are maintained. If you so wish, you can 
exercise your rights to access, rectification and erasure of your data, and 
any other legally held rights, by writing to the sender’s email address or 
***@***.*** ***@***.***>.
This message and, where 
applicable, any attachments are confidential and addressed solely to the 
individual or organization they were sent to.
The UOC has a data protection 
officer who not only supervises the data processing carried out at the 
University, but who will also respond to any questions you may have about 
this data processing. You can contact our data protection officer by 
writing to ***@***.*** ***@***.***>.

**Comment by XaviMartiCamarasa** [comment]
Many many thanks Max for your support on this matter.


On Thu, Nov 24, 2022 at 10:46 PM Jose Javier Marti Camarasa <
***@***.***> wrote:

> Thanks for the advice.
>
> On Thu, Nov 24, 2022 at 10:44 PM Max Halford ***@***.***>
> wrote:
>
>> Right, but machine learning is not about opinions. I suggest you measure
>> the performance of both approaches to be sure.
>>
>> My "opinion" is that recreating a model from scratch is detrimental in
>> that you lose all the information you had accumulated.
>>
>> I think you should distinguish slow drift from brutal drift. If the drift
>> is brutal, then yes you might want to start from scratch. But if the drift
>> is gradual, then you might be ok just having your model learn.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/online-ml/river/discussions/1097#discussioncomment-4232147>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A37HEL6FWXRERD55KLVYXZ3WJ7OUZANCNFSM6AAAAAASKADCRQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 



INFORMACIÓ SOBRE PROTECCIÓ DE DADES DE LA UNIVERSITAT OBERTA DE 
CATALUNYA (UOC)

Us informem que les vostres dades identificatives i les 
contingudes en els missatges electrònics i fitxers adjunts es poden 
incorporar a les nostres bases de dades amb la finalitat de gestionar les 
relacions i comunicacions vinculades a la UOC, i que es poden conservar 
mentre es mantingui la relació. Si ho voleu, podeu exercir el dret a 
accedir a les vostres dades, rectificar-les i suprimir-les i altres drets 
reconeguts normativament adreçant-vos a l'adreça de correu emissora o a 
***@***.*** ***@***.***>.

Aquest missatge i qualsevol 
fitxer que porti adjunt, si escau, tenen el caràcter de confidencials i 
s'adrecen únicament a la persona o entitat a qui s'han enviat.

Així 
mateix, posem a la vostra disposició un delegat de protecció de dades que 
no només s'encarregarà de supervisar tots els tractaments de dades de la 
nostra entitat, sinó que us podrà atendre per a qualsevol qüestió 
relacionada amb el tractament de dades. La seva adreça de contacte és 
***@***.*** ***@***.***>.
INFORMACIÓN SOBRE PROTECCIÓN DE DATOS DE 
LA UNIVERSITAT OBERTA DE CATALUNYA (UOC)
Os informamos de que vuestros 
datos identificativos y los contenidos en los mensajes electrónicos y 
ficheros adjuntos pueden incorporarse a nuestras bases de datos con el fin 
de gestionar las relaciones y comunicaciones vinculadas a la UOC, y de que 
pueden conservarse mientras se mantenga la relación. Si lo deseáis, podéis 
ejercer el derecho a acceder a vuestros datos, rectificarlos y suprimirlos 
y otros derechos reconocidos normativamente dirigiéndoos a la dirección de 
correo emisora o a ***@***.*** ***@***.***>.
Este mensaje y 
cualquier fichero que lleve adjunto, si procede, tienen el carácter de 
confidenciales y se dirigen únicamente a la persona o entidad a quien se 
han enviado.
Así mismo, ponemos a vuestra disposición a un delegado de 
protección de datos que no solo se encargará de supervisar todos los 
tratamientos de datos de nuestra entidad, sino que podrá atenderos para 
cualquier cuestión relacionada con el tratamiento de datos. Su dirección de 
contacto es ***@***.*** ***@***.***>.


UNIVERSITAT OBERTA DE 
CATALUNYA (UOC) DATA PROTECTION INFORMATION
Your personal data and the data 
contained in your email messages and attached files may be stored in our 
databases for the purpose of maintaining relations and communications 
linked to the UOC, and the data may be stored for as long as these 
relations and communications are maintained. If you so wish, you can 
exercise your rights to access, rectification and erasure of your data, and 
any other legally held rights, by writing to the sender’s email address or 
***@***.*** ***@***.***>.
This message and, where 
applicable, any attachments are confidential and addressed solely to the 
individual or organization they were sent to.
The UOC has a data protection 
officer who not only supervises the data processing carried out at the 
University, but who will also respond to any questions you may have about 
this data processing. You can contact our data protection officer by 
writing to ***@***.*** ***@***.***>.

**Comment by MaxHalford** [comment]
No worries @XaviMartiCamarasa, and good luck to you! You're in uncharted waters, so it's normal that you may find it difficult :)

**Comment by XaviMartiCamarasa** [comment]
Thanks, I'll check it out.

On Fri, Nov 25, 2022 at 10:28 AM Max Halford ***@***.***>
wrote:

> River has some Rust code in it. So when you install it from git (via pip),
> you need Rust to compiles these pieces of code. If you try without Rust, it
> will ask you to install Rust.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/online-ml/river/discussions/1097#discussioncomment-4235773>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A37HEL7574ZWCWGF24AYYZ3WKCBB7ANCNFSM6AAAAAASKADCRQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 



INFORMACIÓ SOBRE PROTECCIÓ DE DADES DE LA UNIVERSITAT OBERTA DE 
CATALUNYA (UOC)

Us informem que les vostres dades identificatives i les 
contingudes en els missatges electrònics i fitxers adjunts es poden 
incorporar a les nostres bases de dades amb la finalitat de gestionar les 
relacions i comunicacions vinculades a la UOC, i que es poden conservar 
mentre es mantingui la relació. Si ho voleu, podeu exercir el dret a 
accedir a les vostres dades, rectificar-les i suprimir-les i altres drets 
reconeguts normativament adreçant-vos a l'adreça de correu emissora o a 
***@***.*** ***@***.***>.

Aquest missatge i qualsevol 
fitxer que porti adjunt, si escau, tenen el caràcter de confidencials i 
s'adrecen únicament a la persona o entitat a qui s'han enviat.

Així 
mateix, posem a la vostra disposició un delegat de protecció de dades que 
no només s'encarregarà de supervisar tots els tractaments de dades de la 
nostra entitat, sinó que us podrà atendre per a qualsevol qüestió 
relacionada amb el tractament de dades. La seva adreça de contacte és 
***@***.*** ***@***.***>.
INFORMACIÓN SOBRE PROTECCIÓN DE DATOS DE 
LA UNIVERSITAT OBERTA DE CATALUNYA (UOC)
Os informamos de que vuestros 
datos identificativos y los contenidos en los mensajes electrónicos y 
ficheros adjuntos pueden incorporarse a nuestras bases de datos con el fin 
de gestionar las relaciones y comunicaciones vinculadas a la UOC, y de que 
pueden conservarse mientras se mantenga la relación. Si lo deseáis, podéis 
ejercer el derecho a acceder a vuestros datos, rectificarlos y suprimirlos 
y otros derechos reconocidos normativamente dirigiéndoos a la dirección de 
correo emisora o a ***@***.*** ***@***.***>.
Este mensaje y 
cualquier fichero que lleve adjunto, si procede, tienen el carácter de 
confidenciales y se dirigen únicamente a la persona o entidad a quien se 
han enviado.
Así mismo, ponemos a vuestra disposición a un delegado de 
protección de datos que no solo se encargará de supervisar todos los 
tratamientos de datos de nuestra entidad, sino que podrá atenderos para 
cualquier cuestión relacionada con el tratamiento de datos. Su dirección de 
contacto es ***@***.*** ***@***.***>.


UNIVERSITAT OBERTA DE 
CATALUNYA (UOC) DATA PROTECTION INFORMATION
Your personal data and the data 
contained in your email messages and attached files may be stored in our 
databases for the purpose of maintaining relations and communications 
linked to the UOC, and the data may be stored for as long as these 
relations and communications are maintained. If you so wish, you can 
exercise your rights to access, rectification and erasure of your data, and 
any other legally held rights, by writing to the sender’s email address or 
***@***.*** ***@***.***>.
This message and, where 
applicable, any attachments are confidential and addressed solely to the 
individual or organization they were sent to.
The UOC has a data protection 
officer who not only supervises the data processing carried out at the 
University, but who will also respond to any questions you may have about 
this data processing. You can contact our data protection officer by 
writing to ***@***.*** ***@***.***>.

**Comment by XaviMartiCamarasa** [comment]
Hi @MaxHalford , still facing the same issue.
What I did:
1. Download and install Rust.
2. pip install --> !pip install git+https://github.com/online-ml/river --upgrade
![image](https://user-images.githubusercontent.com/117338671/204037776-db55e18e-b4ac-45e4-aab3-e52f2d65d467.png)

3. Restart the kernel
4. run the code again. Did I anything wrong?

![image](https://user-images.githubusercontent.com/117338671/204037909-6c73f790-7847-4c6a-8c49-70b5bda4d06e.png)
![image](https://user-images.githubusercontent.com/117338671/204037938-e0703d1f-bc01-4841-b17e-723d5ab764df.png)

Kind Regards

**Reply by MaxHalford** [reply]
I have no idea @XaviMartiCamarasa, it's working fine for me. It's difficult for me to debug these kind of issues. However, it does seem like you are not using the new code (I can see that via the error message). So I guess your Python environment has not picked up the fact you installed the latest development version.

**Reply by XaviMartiCamarasa** [reply]
Many thanks @MaxHalford , anyway!
Which is the las commit ingithub? to check what I cloned.

Kind Regards

**Reply by MaxHalford** [reply]
You can see the latest commit ID yourself if you go to the main branch on the River repo.

**Comment by XaviMartiCamarasa** [comment]
I'll do.
Enjoy the weekend Max!

On Sat, Nov 26, 2022 at 11:19 AM Max Halford ***@***.***>
wrote:

> You can see the latest commit ID yourself if you go to the main branch on
> the River repo.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/online-ml/river/discussions/1097#discussioncomment-4242055>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A37HEL74BVF3AVKGN2L7QPTWKHP4BANCNFSM6AAAAAASKADCRQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 



INFORMACIÓ SOBRE PROTECCIÓ DE DADES DE LA UNIVERSITAT OBERTA DE 
CATALUNYA (UOC)

Us informem que les vostres dades identificatives i les 
contingudes en els missatges electrònics i fitxers adjunts es poden 
incorporar a les nostres bases de dades amb la finalitat de gestionar les 
relacions i comunicacions vinculades a la UOC, i que es poden conservar 
mentre es mantingui la relació. Si ho voleu, podeu exercir el dret a 
accedir a les vostres dades, rectificar-les i suprimir-les i altres drets 
reconeguts normativament adreçant-vos a l'adreça de correu emissora o a 
***@***.*** ***@***.***>.

Aquest missatge i qualsevol 
fitxer que porti adjunt, si escau, tenen el caràcter de confidencials i 
s'adrecen únicament a la persona o entitat a qui s'han enviat.

Així 
mateix, posem a la vostra disposició un delegat de protecció de dades que 
no només s'encarregarà de supervisar tots els tractaments de dades de la 
nostra entitat, sinó que us podrà atendre per a qualsevol qüestió 
relacionada amb el tractament de dades. La seva adreça de contacte és 
***@***.*** ***@***.***>.
INFORMACIÓN SOBRE PROTECCIÓN DE DATOS DE 
LA UNIVERSITAT OBERTA DE CATALUNYA (UOC)
Os informamos de que vuestros 
datos identificativos y los contenidos en los mensajes electrónicos y 
ficheros adjuntos pueden incorporarse a nuestras bases de datos con el fin 
de gestionar las relaciones y comunicaciones vinculadas a la UOC, y de que 
pueden conservarse mientras se mantenga la relación. Si lo deseáis, podéis 
ejercer el derecho a acceder a vuestros datos, rectificarlos y suprimirlos 
y otros derechos reconocidos normativamente dirigiéndoos a la dirección de 
correo emisora o a ***@***.*** ***@***.***>.
Este mensaje y 
cualquier fichero que lleve adjunto, si procede, tienen el carácter de 
confidenciales y se dirigen únicamente a la persona o entidad a quien se 
han enviado.
Así mismo, ponemos a vuestra disposición a un delegado de 
protección de datos que no solo se encargará de supervisar todos los 
tratamientos de datos de nuestra entidad, sino que podrá atenderos para 
cualquier cuestión relacionada con el tratamiento de datos. Su dirección de 
contacto es ***@***.*** ***@***.***>.


UNIVERSITAT OBERTA DE 
CATALUNYA (UOC) DATA PROTECTION INFORMATION
Your personal data and the data 
contained in your email messages and attached files may be stored in our 
databases for the purpose of maintaining relations and communications 
linked to the UOC, and the data may be stored for as long as these 
relations and communications are maintained. If you so wish, you can 
exercise your rights to access, rectification and erasure of your data, and 
any other legally held rights, by writing to the sender’s email address or 
***@***.*** ***@***.***>.
This message and, where 
applicable, any attachments are confidential and addressed solely to the 
individual or organization they were sent to.
The UOC has a data protection 
officer who not only supervises the data processing carried out at the 
University, but who will also respond to any questions you may have about 
this data processing. You can contact our data protection officer by 
writing to ***@***.*** ***@***.***>.
